go.mod

```diff
@@ -598,6 +611,13 @@ replace (
 	github.com/openshift/hive => github.com/openshift/hive v1.1.17-0.20220719141355-c63c9b0281d8
 	github.com/openshift/hive/apis => github.com/openshift/hive/apis v0.0.0-20220719141355-c63c9b0281d8
 	k8s.io/cloud-provider-vsphere => github.com/openshift/cloud-provider-vsphere v1.19.1-0.20211222185833-7829863d0558
-	sigs.k8s.io/cluster-api => github.com/openshift/cluster-api v0.0.0-20220303094157-f9c215c4f298
-	sigs.k8s.io/cluster-api-provider-ibmcloud => github.com/openshift/cluster-api-provider-ibmcloud v0.0.0-20221102085410-c1304c80d736
+	sigs.k8s.io/cluster-api-provider-ibmcloud => github.com/openshift/cluster-api-provider-ibmcloud v0.0.0-20231208013118-0fe74f674bf5
 )
+
+replace github.com/openshift/assisted-service => github.com/openshift/assisted-service v1.0.10-0.20240418135216-844e6efd0d1c
+
+replace github.com/openshift/client-go => github.com/openshift/client-go v0.0.0-20221019143426-16aed247da5c
+
+replace sigs.k8s.io/cluster-api => github.com/openshift/cluster-api v0.0.0-20231116115436-03d89f216e0f
```

Comment:
I wonder where this replacement for sigs.k8s.io/cluster-api came from, because it's a different version from the replacement that exists in the fork.

It is pinned there to a specific version of upstream K8s repo. 
https://github.com/openshift/installer-aro/blob/release-4.12-azure/go.mod#L245C2-L245C25

Reply:
I modified this line after running `hack/update-go-module-dependencies.sh`. Before, it read:
```
replace github.com/openshift/cluster-api => github.com/openshift/cluster-api v0.0.0-20231116115436-03d89f216e0f
```
But i got this error message at the end of the script and when running `go mod tidy`:
```
go: github.com/openshift/cluster-api@v0.0.0-20191129101638-b09907ac6668 (replaced by github.com/openshift/cluster-api@v0.0.0-20231116115436-03d89f216e0f): parsing go.mod:
	module declares its path as: sigs.k8s.io/cluster-api
	       but was required as: github.com/openshift/cluster-api
```

Reply:
What I meant is that upstream installer is using:
`sigs.k8s.io/cluster-api => sigs.k8s.io/cluster-api v0.4.5`

They don't seem to be replacing in 4.12 this module with their OpenShift fork anymore and the script did not do the replace either, so I guess it has been in go.mod since previous versions or it was changed manually when we released 4.12

In any case, if the OCP fork for 4.12 worked in your tests, and it's a more recent version with bugfixes, I think we're good.